pinot-controller/src/main/java/org/apache/pinot/controller/api/upload/ZKOperator.java

```diff
@@ -62,52 +66,58 @@ public ZKOperator(PinotHelixResourceManager pinotHelixResourceManager, Controlle
     _controllerMetrics = controllerMetrics;
   }
 
-  public void completeSegmentOperations(String tableNameWithType, SegmentMetadata segmentMetadata,
-      FileUploadType uploadType, @Nullable URI finalSegmentLocationURI, File segmentFile,
-      @Nullable String sourceDownloadURIStr, String segmentDownloadURIStr, @Nullable String crypterName,
-      long segmentSizeInBytes, boolean enableParallelPushProtection, boolean allowRefresh, HttpHeaders headers)
+  public void completeSegmentsOperations(String tableNameWithType, FileUploadType uploadType,
```

Comment:
Consider using different method to handle single segment upload vs batch segment upload. We may extract common part as needed, but modifying the existing flow can be risky. It can cause backward incompatible if any error handling behavior is changed